channeldb/addr_source.go

```diff
@@ -13,3 +13,55 @@ type AddrSource interface {
 	// key.
 	AddrsForNode(nodePub *btcec.PublicKey) ([]net.Addr, error)
 }
+
+// multiAddrSource is an implementation of AddrSource which gathers all the
+// known addresses for a given node from multiple backends and de-duplicates the
+// results.
+type multiAddrSource struct {
+	sources []AddrSource
+}
+
+// NewMultiAddrSource constructs a new AddrSource which will query all the
+// provided sources for a node's addresses and will then de-duplicate the
+// results.
+func NewMultiAddrSource(sources ...AddrSource) AddrSource {
+	return &multiAddrSource{
+		sources: sources,
+	}
+}
+
+// AddrsForNode returns all known addresses for the target node public key.
+//
+// NOTE: this implements the AddrSource interface.
```

Comment:
could mention that we are using the union of the sets instead of the intersection. And also have a question here - if the node is not found in either DB, does it mean there's a bug somewhere? If it's not in the link db, think the node cannot be used for sending htlcs. when it's not in the graph db, it won't be used for pathfinding.